ctapipe/core/container.py

```diff
@@ -59,10 +59,16 @@ def __new__(cls, name, bases, dct):
                 for k, v in b.fields.items():
                     dct['fields'][k] = v
 
-        for k in items:
+        for k in field_names:
```

Comment:
If you like, I believe `for name in field_names` might be easier to read than `k`